pkg/client/retrieve.go

```diff
@@ -34,10 +32,19 @@ import (
 	"k8s.io/client-go/tools/remotecommand"
 )
 
-func (c *SonobuoyClient) RetrieveResults(cfg *RetrieveConfig) (io.Reader, error) {
+var tarCommand = []string{
+	"/usr/bin/env",
+	"bash",
+	"-c",
+	fmt.Sprintf("tar cf - %s/*.tar.gz", config.MasterResultsPath),
+}
+
+func (c *SonobuoyClient) RetrieveResults(cfg *RetrieveConfig) (io.Reader, <-chan error) {
```

Comment:
This can't really be blocking if we want to keep the io.Reader contract. In the correct case, the error code isn't available until after execution is completed. In the error case, it could be an arbitrary amount of time (10 or more seconds) until we get the report of the status code.

Reply:
I don't mind blocking for (X) with a ctx arg provided, that standard go.  Otherwise, we need to revisit the whole interface.